pkg/services/ngalert/store/image.go

```diff
@@ -83,6 +92,35 @@ func (st DBstore) SaveImage(ctx context.Context, img *Image) error {
 	})
 }
 
+func (st *DBstore) GetURL(ctx context.Context, token string) (*url.URL, error) {
+	img, err := st.GetImage(ctx, token)
+	if err != nil {
+		return &url.URL{}, err
+	}
+	return url.Parse(img.URL)
+}
+
+func (st *DBstore) GetData(ctx context.Context, token string) (io.ReadCloser, error) {
+	// TODO: Should we support getting data from image.URL? One could configure
+	// the system to upload to S3 while still reading data for notifiers like
+	// Slack that take multipart uploads.
```

Comment:
Seems like a great idea. After 9.0? 😄Seems like this would maybe need some longer-run testing to make sure we don't hit problems with timeouts on chunks somewhere along that chain.

Reply:
Agreed, after 9.0